### ISSUE TITLE
Registry editor hangs on startup

### ISSUE THREAD

**Issue by DanielSank** [maffoo/scalabrad#23]
Brian starts the scalabrad manager from bintray and sees

```
12:42:39.612 [main] INFO  org.labrad.manager.Manager$ - registry location: C:\Us
ers\Brian\.labrad\registry.sqlite, format=sqlite
12:42:41.087 [main] INFO  org.labrad.manager.Listener - now accepting labrad con
nections on port 7682
```

He then starts the registry editor version 1.0.0 and the UI hangs showing no directories.

**Comment by DanielSank** [comment]
Brian, what are the values of the following environment variables:
1. `LABRADHOST`
2. `LABRADPORT`

Do you have a `LABRADPASSWORD` set? (don't post it)

What version of the manager are you running?

**Comment by maffoo** [comment]
please do not post your LABRADPASSWORD on this public page :)

**Comment by DanielSank** [comment]
Grah! Fixed.

**Comment by Brian-Ballard** [comment]
LABRADHOST = localhost
LABRADPASSWORD = (blank)
LABRADPORT = 7682

**Comment by maffoo** [comment]
what version of scalabrad are you using?

**Comment by Brian-Ballard** [comment]
0.3.3

**Comment by maffoo** [comment]
is it possible the password environment variable is not set the same for both the registry editor and the shell where you launched scalabrad?

**Comment by maffoo** [comment]
also, I just want to clarify, is the UI hanging or just not showing any directories? do you expect there to be data in the registry?

**Comment by Brian-Ballard** [comment]
yes that is possible that the variable is not the same for the registry editor and labrad.  How do i get them to conform.

**Comment by DanielSank** [comment]
Assuming you set the environment variables globally (i.e. through the control panel) just close everything and restart.

**Comment by Brian-Ballard** [comment]
I don't expect to have data in the registry as this is a new setup.  I think it is hung because i cannot add folders or perform other tasks

**Comment by maffoo** [comment]
ok, got it. if it's hung that _probably_ means the password is wrong and it is failing to connect. We do have a newer version of the registry editor that we could send, but I think the only change is a new icon; we're working on replacing it entirely as part of our delphi purge.

**Comment by DanielSank** [comment]
Ok, so @Brian-Ballard, make sure the environment variables are set via the control panel (i.e. not just manually in the command line), close the manager and registry editor, and then restart them.

**Comment by Brian-Ballard** [comment]
ok closed and restarted

**Comment by Brian-Ballard** [comment]
Ok wait.  Which control pannel and which "Manager"

**Comment by DanielSank** [comment]
and?

**Comment by Brian-Ballard** [comment]
What or where is the control pannel?

**Comment by DanielSank** [comment]
By "control panel" I mean the thing in Windows that lets you set environment variables. I mentioned this because you can also set environment variables in a command shell, but if you do it that way the variables only apply to _that session_ and won't affect other programs.

By "manager" I mean the LabRAD manager, the thing you downloaded from bintray.

**Comment by Brian-Ballard** [comment]
Okay I am with you now

**Comment by DanielSank** [comment]
So, did it work?

**Comment by Brian-Ballard** [comment]
No.  I don't get to prompts for those values anymore.  But the behavior is the same

**Comment by DanielSank** [comment]
@Brian-Ballard is there a file at `C:\Users\Brian\.labrad\registry.sqlite`?

**Comment by Brian-Ballard** [comment]
yes

**Comment by maffoo** [comment]
@Brian-Ballard, do you have python set up on your machine with pylabrad? If so are you able to connect to the manager from python?

**Comment by Brian-Ballard** [comment]
yes and yes

**Comment by maffoo** [comment]
Hm, could you try putting some data in the registry with python and then restarting the registry editor? I don't know if we've ever tested looking at an empty registry...

**Comment by Brian-Ballard** [comment]
Sure.  Can you give me a snippet.  I have a python shell open and i did the usual import labrad and cxn = labrad.connect()  what do i do from there to put some data in the registry?

**Comment by maffoo** [comment]
``` python
>>> cxn.registry.mkdir('test') # create a directory
>>> cxn.registry.set('x', 'this is a test') # add a key
```

**Comment by Brian-Ballard** [comment]
Looks like those values were entered into the registry.sqllite but the registry editor is exibiting the same behavior.  It does this after it is restarted and it and labrad are restarted.

**Comment by DanielSank** [comment]
> Looks like those values were entered into the registry.sqllite

Please be very specific. How do you know the values were entered? The file size changed? You read them back?

**Comment by Brian-Ballard** [comment]
I pulled registry.sqllite up in a text editor and seached for the text "test" and "this is a test"  Both were found

**Comment by DanielSank** [comment]
Check and see if there's a config file sitting in the directory with the registry editor. If there is, check that there isn't a bad password or something like that.

**Comment by Brian-Ballard** [comment]
I can also read them back in the python interface.

**Comment by Brian-Ballard** [comment]
There is no config file

**Comment by DanielSank** [comment]
Please describe how you set up the registry editor. Presumably you have an .exe file in a directory somewhere and just double-clicked it to run. What else is in that directory?

**Comment by Brian-Ballard** [comment]
ghz_fpga_server.py and direct ethernet execuable.

**Comment by maffoo** [comment]
The registry editor doesn't use a config file; it just pulls params from environment vars. We should send you the latest version of the registry editor that we have; I had thought the only change was to update the icon, but there may be some other slight differences. I think the latest is version 1.0.2.

**Comment by DanielSank** [comment]
Sent 1.0.2 to Brian and it works. That is very strange to me, but c'est la vie.

**Comment by maffoo** [comment]
Oh right, here's the fix in version 1.0.2: https://github.com/martinisgroup/labrad-manager-delphi/commit/0cd2f3518970246cff6a38ec084513188c1410fe

The issue was that the delphi manager defined the settings with the lower-case O like 'Notify on Change' and scalabrad copied that. However, the delphi manager actually used case-insensitive comparison to look up setting names, so the different case didn't cause an issue.

I remember discussing this at the time with some folks (at least with evan since he reviewed the registry editor change).
